code/game/objects/items/tcg/tcg_machines.dm

```diff
@@ -218,3 +218,176 @@ GLOBAL_LIST_EMPTY(tcgcard_machine_radial_choices)
 
 /obj/effect/overlay/card_summon
 	mouse_opacity = 0
+
+/obj/machinery/trading_card_button
+	name = "mana control panel"
+	desc = "a set of buttons that lets you keep track of your mana when playing Tactical Game Cards."
+	icon = 'icons/obj/toys/tcgmisc.dmi'
+	icon_state = "mana_buttons"
+	use_power = NO_POWER_USE
+	
+	///Reference to the display panel generated by this button.
+	var/obj/effect/decal/trading_card_panel/display_panel_ref
+	///Typepath of the display panel generated.
+	var/display_panel_type = /obj/effect/decal/trading_card_panel
+	///Where the panel will be spawned in relation to the button on the X axis.
+	var/panel_offset_x = 1
+	///Where the panel will be spawned in relation to the button on the Y axis.
+	var/panel_offset_y = 0
+
+GLOBAL_LIST_EMPTY(tcgcard_mana_bar_radial_choices)
+
+/obj/machinery/trading_card_button/Initialize(mapload)
+	. = ..()
+	display_panel_ref = new display_panel_type(locate(x + panel_offset_x, y + panel_offset_y, z))
+
+/obj/machinery/trading_card_button/Destroy()
+	qdel(display_panel_ref)
```

Comment:
```suggestion
	QDEL_NULL(display_panel_ref)
```
QDELL_NULL also empties the variable. If we leave lingering references we can force a harddel which is bad